libr/core/cmd_zign.c

```diff
@@ -80,6 +80,32 @@ static void cmd_zign_init(RCore *core) {
 	DEFINE_CMD_DESCRIPTOR (core, zs);
 }
 
+static bool addFcnHash(RCore *core, RAnalFunction *fcn, const char *name) {
+	if (!core || !fcn || !name) {
+		return false;
+	}
+	int len = r_anal_fcn_realsize (fcn);
+
+	if (!r_io_is_valid_offset (core->io, fcn->addr, 0)) {
+		eprintf ("error: cannot read at 0x%08"PFMT64x"\n", fcn->addr);
+		return false;
+	}
+
+	ut8 *buf = malloc (len);
+	if (!buf) {
+		return false;
+	}
+
+	(void)r_io_read_at (core->io, fcn->addr, buf, len);
```

Comment:
this is a bad idea, imho you should only keep bb hash, because functions doesnt need to be linear, and for those cases you can find functions of several megabytes because they are composed by other data or other functions code. so, if the function is linear, the bbhash should be the same as the fcnhash, because you should be hashing the bbs sorted by offset.

and maybe this linear function works for some bins, but i can give you some bins that will make this fail hard